### DISCUSSION TITLE
What would you like to see from this project?

### DISCUSSION THREAD

**Discussion by brandonmpetty** [brandonmpetty/Doxa#34] [Polls]
I have received a lot of positive feedback from people using this project.
I would love to get some feedback on what to work on next. 
Below are a few ideas that have come from discussions I've had with members of the community.
If there is something you'd like to see that is not on the list, feel free to mention it.

**Comment by flawrist** [comment]
Windows 7 x64 user here. I'd like to see your library as a single `.exe` command-line tool to process files. For example, like [nQuantCpp](https://github.com/mcychan/nQuantCpp) by @mcychan  does. This is the only way to bring your invaluable work to ordinary people unfamiliar with the intricacies of compiling, but whose burden you can certainly ease.

**Reply by brandonmpetty** [reply]
@sergeevabc , this comment has been on my mind for a long time now.  You are the first I am sharing it with, that this project finally has a cli tool: https://github.com/brandonmpetty/Doxa/pull/57

The utility is called BinBench, a Python script that exposes all binarization and grayscale algorithms in this library as well as all performance metrics.  This is a drop-in replacement for the DIBCO executables, but this one runs cross platform, is fully open source, and is extremely performant (though the raw C++ is still faster).

Anyway, I hope you like it.  This also comes with full parameter documentation on each algorithm as well.

-Thanks again

**Comment by brandonmpetty** [comment]
I have recently merged in AdOtsu and just put together a first pass at Matlab bindings and will see if I can get that merged soon-ish.

I may take a stab at Pseudo F-Measure next: https://github.com/brandonmpetty/Doxa/issues/32

@sergeevabc - After I add Pseudo F-Measure, I will generate a tool to do exactly what you are asking.
At that point, it will have everything needed to replace the DIBCO performance calculation binary, but with the addition of automatically generating binarized output.  I am thinking something like "doxa-cli".

I am curious though, would you rather an .exe, or a web app that would replace the Doxa demo: https://brandonmpetty.github.io/Doxa/WebAssembly

The web app could be updated that would let you upload any file(s) you wanted, with full configurability.  Let me know what you think would be more useful.

**Reply by brandonmpetty** [reply]
Locally, I have Doxa producing all of the performance metrics that the DIBCO tool produces.

I will work on building a DIBCO_metrics.exe replacement that is fully open source, works on Windows/MacOS/Linux, and has the potential to include out of the box Binarization algorithms like Otsu and Sauvola so that anyone can reproduce the results derived from the DIBCO papers. I am excited to finally get this working.

**Reply by brandonmpetty** [reply]
#50 Matlab Bindings were added
#53 v0.9.5 includes Pseudo F-Measure and Grayscale algorithms across all bindings + SIMD support

**Comment by brandonmpetty** [comment]
Every on this list has been knocked out!

- [x] MATLAB bindings
- [x] Exposing grayscale algorithms to all bindings
- [x] AdOtsu binarization algorithm /w grid optimization and multi-scale
- [x] Pseudo F-Measure across all bindings
- [x] SIMD: SSE2, ARM NEON, WASM SIMD 128

I appreciate everyone's support and patience.  It has taken a while to get this far, but the project is still active and alive.
